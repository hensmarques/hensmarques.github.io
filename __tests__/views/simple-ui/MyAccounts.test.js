import React from 'react';
import { MyAccounts } from '../../../src/views/simple-ui/MyAccounts';

describe('<MyAccounts/>', () => {
  const loadTransactions = jest.fn();
  const getTransactionsList = jest.fn();
  const getCompletedTransactionsList = jest.fn();
  const loadData = jest.fn();
  it('renders and matches the snapshot', () => {
    const wrapper = shallow(
      <MyAccounts
        getTransactionsList={getTransactionsList}
        loadTransactions={loadTransactions}
        getCompletedTransactionsList={getCompletedTransactionsList}
        loadData={loadData}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });

  describe('transaction modal integration', () => {
    const mockState = {
      modal: 'deposit',
      currency: 'USD'
    };
    let wrapper;

    beforeEach(() => {
      wrapper = shallow(
        <MyAccounts
          getTransactionsList={getTransactionsList}
          loadTransactions={loadTransactions}
          getCompletedTransactionsList={getCompletedTransactionsList}
          loadData={loadData}
        />
      );
    });

    it('renders the TransactionModal', () => {
      expect(wrapper.find('Connect(TransactionModal)').exists()).toBe(false);
      wrapper.setState(mockState);
      expect(wrapper.find('Connect(TransactionModal)').exists()).toBe(true);
      expect(
        wrapper.find('Connect(TransactionModal)').props().transactionType
      ).toBe(mockState.modal);
      expect(wrapper.find('Connect(TransactionModal)').props().currency).toBe(
        mockState.currency
      );
    });

    it('hides the TransactionModal', () => {
      wrapper.setState(mockState);
      expect(wrapper.find('Connect(TransactionModal)').exists()).toBe(true);
      wrapper
        .find('Connect(TransactionModal)')
        .props()
        .showModal();
      expect(wrapper.find('Connect(TransactionModal)').exists()).toBe(false);
    });
  });
});
