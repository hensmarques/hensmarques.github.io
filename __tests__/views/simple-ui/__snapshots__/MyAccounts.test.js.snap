// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MyAccounts/> renders and matches the snapshot 1`] = `
<Connect(SimpleUiWrapper)
  description=""
  pageClass="my-accounts"
  pageHeader=""
  title="My Accounts"
>
  <Connect(ErrorBoundary)>
    <Connect(CurrentBalances)
      modal=""
      showModal={[Function]}
    />
  </Connect(ErrorBoundary)>
  <div
    className="component my-accounts-table"
    id="my-accounts-table"
  >
    <Connect(ErrorBoundary)>
      <Connect(MyAccountsTablesContainer) />
    </Connect(ErrorBoundary)>
  </div>
</Connect(SimpleUiWrapper)>
`;
